### PR TITLE
Add support for namespace

### PR DESCRIPTION
**Pull request by homalozoa**
Signed-off-by: Homalozoa <xuhaiwang@xiaomi.com>

### CODE REVIEW

**Comment by fmrico** [comment]
Hi @paeunt 

Thanks for your contributions. LGTM, but I wonder if you could add some tests to [rclcpp_cascade_lifecycle_test.cpp](https://github.com/fmrico/cascade_lifecycle/blob/9ab9f225a2a40147def7d0de8529a747d4b3ad02/rclcpp_cascade_lifecycle/test/rclcpp_cascade_lifecycle_test.cpp) in order to cover also this situation.

Best

**Comment by homalozoa** [comment]
No problem, I'll add test for namespace later.

**Review by fmrico — CHANGES_REQUESTED** [review]
Please, don't change `activations_managing_late_joining`. You are mixing no-namespaced nodes with namespaced nodes.

**Comment by homalozoa** [comment]
Sorry, my mistake. I already fix it.

**Comment by homalozoa** [comment]
The CI is OK in my fork : [master](https://github.com/paeunt/cascade_lifecycle/actions/workflows/master.yaml)
But the versions are conflict in https://github.com/fmrico/cascade_lifecycle/pull/4/conflicts, what version shoud it be? 0.0.6? or 0.0.8?

**Review by fmrico — APPROVED** [review]

File: rclcpp_cascade_lifecycle/CHANGELOG.rst [hunk 1]
```diff
@@ -1,6 +1,10 @@
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 Changelog for package rclcpp_cascade_lifecycle
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+0.0.6 (2021-06-03)
```
Comment (fmrico) [inline rclcpp_cascade_lifecycle/CHANGELOG.rst]: Don't modify this file, please

File: rclcpp_cascade_lifecycle/package.xml [hunk 2]
```diff
@@ -2,7 +2,7 @@
 <?xml-model href="http://download.ros.org/schema/package_format3.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
 <package format="3">
   <name>rclcpp_cascade_lifecycle</name>
-  <version>0.0.5</version>
+  <version>0.0.6</version>
```
Comment (fmrico) [inline rclcpp_cascade_lifecycle/package.xml]: Don't modify this file, please, 0.0.5 is fine here.

File: .github/workflows/master.yaml [hunk 3]
```diff
@@ -17,19 +17,19 @@ jobs:
       fail-fast: false
     steps:
       - name: Setup ROS 2
-        uses: ros-tooling/setup-ros@0.1.1
+        uses: ros-tooling/setup-ros@v0.2
```
Comment (fmrico) [inline .github/workflows/master.yaml:20]: Thanks for updating this. Maybe 0.2.1 is better, right?

File: .github/workflows/master.yaml [hunk 4]
```diff
@@ -17,19 +17,19 @@ jobs:
...
         with:
           required-ros-distributions: foxy
       - name: build and test
-        uses: ros-tooling/action-ros-ci@0.1.0
+        uses: ros-tooling/action-ros-ci@v0.2
```
Comment (fmrico) [inline .github/workflows/master.yaml:24]: 0.2.1 is also better here, right?

**Review by fmrico — CHANGES_REQUESTED** [review]

**Comment by fmrico** [comment]
Don't worry by now about versions. I will manage this in the packaging process

CI has passed here, so address my last comments, and we are ready to merge :wink:

**Comment by homalozoa** [comment]
OK. I'll edit them. BTW, should I change foxy to galactic ? @fmrico

**Comment by fmrico** [comment]
I have to think about it because Galactic is here now. 

I am maintaining more packages, and I should do the same for all of them.

**Comment by homalozoa** [comment]
That sounds good. So I'll keep `foxy` in this PR.

File: .github/workflows/master.yaml (hunk 3 above)
Comment (homalozoa) [reply .github/workflows/master.yaml:20]: seems 0.2.1 is not exist... ref:https://github.com/paeunt/cascade_lifecycle/runs/2735240985

Comment (homalozoa) [reply .github/workflows/master.yaml:20]: As shown in https://github.com/ros-tooling/action-ros-ci/releases, there is no v0.2.1, only v0.2 for it. I'll change it back.

**Review by fmrico — APPROVED** [review]

**Comment by fmrico** [comment]
Merged!!!

Thanks

**Pull request merged**